compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -178,6 +178,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         let ty = match pat.kind {
             PatKind::Wild => expected,
+            PatKind::Never => expected, // FIXME(never_patterns): check the type is uninhabited
```

Comment:
I would somewhat prefer this gets done in this PR, but I guess it doesn't have to.

Reply:
I don't know how to do that actually. I'm not aware of anywhere else where "T is empty" is a typeck constraint; would you know how to do that? I don't think `inhabited_predicate` can be used during typeck either

Reply:
IDK if it's possible. We may need to defer it until later. In that case, maybe update the FIXME to say "investigate if it's possible to check that the type is uninhabited", or something?